src/Solver.cpp

```diff
@@ -1496,6 +1485,26 @@ void Solver::initializeSettings()
 
     env->settings->createSetting("ResultPath", "Output", empty, "The path where to save the result information", true);
 
+    // Need to create the modeling systems temporary to get their settings
+    {
+        auto modelingSystem = std::make_shared<ModelingSystemOSiL>(env);
+        modelingSystem->augmentSettings(env->settings);
+    }
```

Comment:
In Ipopt, the augmentSettings function (called RegisterOptions there) is static, so you won't have to create an object first.

Reply:
Good idea. Fixed this.